### ISSUE TITLE
Potential Error in Chamfer Distance Calculation

### ISSUE THREAD

**Issue by samuelss1996** [fwilliams/point-cloud-utils#89]
#### Description

I noticed a potential issue in the calculation of the Chamfer Distance within the codebase. Specifically, the current implementation sums the mean distances rather than taking the average.

#### Current Implementation

In the current implementation, the Chamfer Distance is calculated as follows:

https://github.com/fwilliams/point-cloud-utils/blob/387b84f658efc10464e7c3cf5bed5f54c1c711e3/point_cloud_utils/__init__.py#L115

#### Issue

The Chamfer Distance is typically defined as the average of the mean distances, not the sum. This is in fact, what is written in the project's documentation:

$$ \text{chamfer}(P_1, P_2) = \frac{1}{2n} \sum_{x \in P_1} |x - \text{NN}(x, P_2)| + \frac{1}{2m} \sum_{x \in P_2} |x - \text{NN}(x, P_1)| $$

Source: https://github.com/fwilliams/point-cloud-utils/blob/master/docs/docs/sections/shape_metrics.md#chamfer-distance

The current method calculates the sum of the mean distances, thus performing the following computation instead:

$$ \text{chamfer}(P_1, P_2) = \frac{1}{n} \sum_{x \in P_1} |x - \text{NN}(x, P_2)| + \frac{1}{m} \sum_{x \in P_2} |x - \text{NN}(x, P_1)| $$

(notice the change in the denominators with respect to the original equation)

#### Proposed Fix

The correct calculation should take the average of these distances:

```python
cham_dist = (np.mean(dists_x_to_y) + np.mean(dists_y_to_x)) / 2
```

**Comment by daniyar15** [comment]
I fully agree with the posted issue. The incorrectness of the current implementation of Chamfer Distance can be tested using the following snippet:

```
import point_cloud_utils as pcu
import numpy as np

p1 = np.array([[1, 0, 0]], dtype=np.float32)
p2 = np.array([[0, 2, 0]], dtype=np.float32)

# we have only point for each point cloud, so the formula for chamfer distance reduces to:
expected_cd = np.linalg.norm(p1[0, ] - p2[0, ]) / 2 + np.linalg.norm(p2[0, ] - p1[0, ]) / 2
# calculate chamfer distance using the library
cd = pcu.chamfer_distance(p1, p2)
print("Expected Chamfer Distance:", expected_cd)
print("Calculated Chamfer Distance:", cd)
```
which has the following output:
```
Expected Chamfer Distance: 2.2360680103302
Calculated Chamfer Distance: 4.472136
```
The current implementation actually computes 2 times of the actual Chamfer Distance if we use the formula from the docs. Proposed fix would work, but we also don't have to compute means there because `dists_x_to_y` and `dists_y_to_x` are already means over distances. The following line 
https://github.com/fwilliams/point-cloud-utils/blob/a5ce154b32b9eaae8639a2aa936db3c600f25da9/point_cloud_utils/__init__.py#L115
should be replaced with
```
cham_dist = (dists_x_to_y + dists_y_to_x) / 2.0
```
